tests/unit/test_main.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import sys
+
+from unittest.mock import patch, Mock
+
+from auditwheel.main import main
+
+
+def test_non_linux_platform(monkeypatch, capsys):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "darwin")
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 1
+    captured = capsys.readouterr()
+    assert captured.out == "Error: This tool only supports Linux\n"
+
+
+@patch("auditwheel.main.argparse.ArgumentParser")
+@patch("auditwheel.main.logging.basicConfig")
+def test_verbose_logging_non_verbose(basic_cfg_mock, parser_mock, monkeypatch):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "linux")
+    args = Mock()
+    args.verbose = 0
+    args.func.return_value = 0
+    parser_mock.return_value.parse_args.return_value = args
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 0
+    basic_cfg_mock.assert_called_once_with(level=logging.INFO)
+
+
+@patch("auditwheel.main.argparse.ArgumentParser")
+@patch("auditwheel.main.logging.basicConfig")
+def test_verbose_logging_verbose(basic_cfg_mock, parser_mock, monkeypatch):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "linux")
+    args = Mock()
+    args.verbose = 1
+    args.func.return_value = 0
+    parser_mock.return_value.parse_args.return_value = args
+
+    # WHEN
+    retval = main()
+
+    # THEN
+    assert retval == 0
+    basic_cfg_mock.assert_called_once_with(level=logging.DEBUG)
+
+
+def test_help(monkeypatch, capsys):
+    # GIVEN
+    monkeypatch.setattr(sys, "platform", "linux")
```

Comment:
Here, you're monkey patching the platform again. If the test if relevant (not sure but why not), then just skip the test on non linux platforms. This way, there's no correlation between the platform check and this test.

Reply:
This test checks that not providing any arguments will cause the help message to be displayed. The definition of a unit test is that it tests the SUT independent of its environment. If I let the platform check go through this would not be a real unit test as it would become environment dependent. 

Besides, I think it makes sense to validate that changes did not break the CLI logic on a non-Linux machine as well. I don't see why patching the platform is an issue here. Unless we stop supporting Linux, it's unlikely that this code will need to change.

Reply:
I guess my point is that, if I ask someone to write a unit test for exactly what you said "This test checks that not providing any arguments will cause the help message to be displayed.", no one would ever monkeypatch the platform. That's reason enough to skip. Furthermore, if the check changes in CLI to use anything else than `sys.platform`, this test will fail and need change.

There are only 2 tests left, but before you forced push there would have been 3 places to change, maybe unlikely but still. I would at least define a fixture to keep this platform monkeypatch reusable.
I would skip it using:
```
skip_unsupported = pytest.mark.skipif(sys.platform != 'linux',
                                      reason="platform not supported")

@skip_unsupported
def test_help(monkeypatch, capsys):
```

or monkeypatch it:
```
@pytest.fixture
def _mock_platform_check(monkeypatch):
    monkeypatch.setattr(sys, "platform", "linux")


mock_platform_check = pytest.mark.usefixtures("_mock_platform_check")

@mock_platform_check
def test_help(monkeypatch, capsys):
```
that way, no need to modify everywhere in case of a change in CLI implementation.

The `skip_unsupported ` is at least not dependent on the usage of `sys.platform` in CLI but would fail to skip if the check in CLI changes to only allow x86/x86_64 architectures for example. Also, it'll still skip the tests if darwin/windows gets supported.

The `mock_platform_check` is still dependent on CLI check implementation, test will fail on other architectures if the check in CLI changes to only allow x86/x86_64 architectures for example.

A more robust way to mock or skip would be to have a function in the main module that does the platform check. It could be mocked or used for skipping in a more robust way.
```
def _platform_supported():
    if sys.platform != 'linux':
        print('Error: This tool only supports Linux')
        return False
    return True


def main():
    if not _platform_supported():
        return 1
```
This is probably taking a sledgehammer to crack a nut

Reply:
> I guess my point is that, if I ask someone to write a unit test for exactly what you said "This test checks that not providing any arguments will cause the help message to be displayed.", no one would ever monkeypatch the platform.

I respectfully disagree. A unit test by definition is a test to validate functionality *in isolation*. If a unit test's result depends on what platform runs, it is not a unit test. Unit tests are supposed to test that the logic in the *unit under test* is correct and disregard all external factors (e.g. no external I/O or calls to anything which might do I/O should be allowed). The fact that often times such tests need to know implementation details in order to patch out external calls is inevitable and it becomes largely subjective to decide how many implementation details are we OK to accept in these tests. If the code is structured well, little mocking should be necessary, but this is not always the case and minimal changes to unit tests due to implementation changes are perfectly fine in my opinion.